localstack/localstack
# remove legacy runtime in localstack/service/infra.py

## Motivation

This is a PR to remove the code that was deprecated by #10942.

I left `signal_supervisor_restart()` in the legacy module (it's still referenced in ext), but added a TODO.

Considering this a minor change since we never documented `LEGACY_RUNTIME` and have, to my knowledge, not needed it so far.


## Changes

* `infra.py` has been completely removed in favor of the new runtime framework


## Testing

One thing needs manual testing, namely the kill function of the health endpoint
* run `python -m localstack.dev.run -e DEBUG=1`
* observe that `curl -X POST -H 'Content-Type: application/json' localhost.localstack.cloud:4566/_localstack/health -d '{"action": "kill"}'` still works
